### DISCUSSION TITLE
Allow ReplayGain to be calculated using RSGain

### DISCUSSION THREAD

**Discussion by haakobja** [beetbox/beets#4630] [Ideas]
Hi,
I was recently introduced to the [RSGain application](https://github.com/complexlogic/rsgain) for calculating ReplayGain values, which is currently in use by the newest ReplayGain plugin to MusicBrainz Picard.

According to the GitHub repository it's able to calculate ReplayGain for the [following formats](https://github.com/complexlogic/rsgain):

>
>Format | Supported File Extension(s)
>-- | --
>Audio Interchange File Format (AIFF) | .aiff
>Free Lossless Audio Codec (FLAC) | .flac
>Monkey's Audio | .ape
>MPEG-1 Audio Layer II (MP2) | .mp2
>MPEG-1 Audio Layer III (MP3) | .mp3
>MPEG-4 Audio (AAC, ALAC)¹ | .m4a
>Musepack (MPC)² | .mpc
>Ogg (Vorbis, Speex, FLAC) | .ogg, .oga, .spx
>Opus | .opus
>Waveform Audio File Format (WAV) | .wav
>Wavpack | .wv
>Windows Media Audio (WMA) | .wma

What do you guys think? Is it worth looking into it as a replacement for mp3gain, or is it enough to use the gstreamer or ffmpeg approaches?

**Comment by sampsyo** [comment]
Seems interesting! I'd be particularly interested to know whether you see any specific advantages over the FFmpeg backend, which is also a "drop-in" command-line utility. Or perhaps it's just one more option?

**Reply by haakobja** [reply]
I look at it as one more option because it depends on FFmpeg as well. So I guess it can be used if someone doesn't want to use GStreamer or FFmpeg directly, and just wants a command line tool to do it for them.

It does support more audio formats than mp3gain or aacgain, so if we were to compare it with them, I would say it's a better replacement. But for replacing ffmpeg, I don't know..

**Reply by sampsyo** [reply]
FWIW, the current ffmpeg backend currently uses ffmpeg as a command-line tool also (it doesn't link the ffmpeg library or anything). So hopefully that is pretty easy to install.

**Comment by djspacewhale** [comment]
I was just wondering if rsgain as a backend was possible - I'm bumping to show this would still be desired.

**Comment by daniel-l** [comment]
I'd love to have rsgain as a backend.

**Comment by teadoku** [comment]
I also wanted `rsgain` as a backend, I've opened a PR so hopefully it will be added soon: https://github.com/beetbox/beets/pull/6948

**Reply by carks32** [reply]
I pray it sorts out! I kind of love rsgain.
